### ISSUE TITLE
Spock tests are not run by Maven on Kotlin project

### ISSUE THREAD

**Issue by nik-lazer** [micronaut-projects/micronaut-test#90]
I have generated Kotlin project with Spock feature and Maven as a build tool:
mn create-app lan.test.micronaut.spock --lang=kotlin -b=maven --features spock

The result project can be seen here
https://github.com/nik-lazer/micronaut-kotlin-spock

Desirable behavior:

1. Project is generated and can be compiled
2. Simple Spock test can be triggered via maven build

Actual behavior:

1. Project does not complie, because ${jdk.version} in Maven was set to 12 despite the fact that gmaven plugin can work with bytecode from 1.8 and not later:
![micronaut-kotlin-spock-bytecode](https://user-images.githubusercontent.com/1555606/67928066-32292180-fbcb-11e9-8b66-3440d3fb17dc.JPG)

 This was fixed by commit https://github.com/nik-lazer/micronaut-kotlin-spock/commit/e562f05d9e0daac59abd43ed72d1d7aa5812b2a6

2. pom file contains two declarations of surefire plugin, causing warning as a result 
![micronaut-kotlin-spock-malformed-pom](https://user-images.githubusercontent.com/1555606/67928132-571d9480-fbcb-11e9-8079-ce37f1bd9014.JPG)

3. Maven build runs no tests and reports that zero tests were triggered. This happens despite the fact that HelloSpec.groovy is in the project and can be run from IDE. Here I did not manage to find a solution so far and have to ask for support.
![micronaut-kotlin-spock-notests](https://user-images.githubusercontent.com/1555606/67928219-86cc9c80-fbcb-11e9-948f-f73329a086bd.JPG)

Has the scheme Kotlin-Maven-Spock ever been tested or should I change it somehow?

**Comment by nik-lazer** [comment]
Actually in case Java+Spock+Maven issues are almost the same (except duplicated surefire in pom):
1. Out of the box Maven build fails and pom has to be edited fro eliminating of this
2. No tests run (exactly as in case with Kotlin+Spock)

The sample project is here:
https://github.com/nik-lazer/micronaut-java-spock

**Comment by nik-lazer** [comment]
The issue with tests looks similar to https://github.com/micronaut-projects/micronaut-core/issues/439

**Comment by stevesea** [comment]
having the same trouble. To reproduce: 

```
mn create-app --build maven --lang java -f spock myproj
cd myproj
mn create-test mytest
mvn clean test
```

The test is compiled and is in target/test-classes , but the Spock test isn't run.

**Comment by benrhine** [comment]
The above mentioned issues will be fixed in a future release once PR https://github.com/micronaut-projects/micronaut-profiles/pull/201 is merged

**Comment by alvarosanchez** [comment]
Unfortunately, https://github.com/micronaut-projects/micronaut-profiles/pull/201 doesn't seem to fix the issues mentioned in this PR.

**Comment by alvarosanchez** [comment]
@stevesea @nik-lazer the Java + Maven + Spock combination is being addressed in #109
